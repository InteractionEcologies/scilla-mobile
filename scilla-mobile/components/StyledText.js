import React from 'react';
// import { Text } from 'react-native';
import { Text, Label } from "native-base"; 
import Fonts from "../constants/Fonts";

const FONT_FAMILY = "space-mono";

export class MonoText extends React.Component {
  render() {
    return <Text {...this.props} style={[this.props.style, { fontFamily: FONT_FAMILY }]} />;
  }
}

export class AppText extends React.Component {
  render() {
    return <Text {...this.props} style={[this.props.style, { fontFamily: Fonts.OpenSans }]} />;
  }
}
