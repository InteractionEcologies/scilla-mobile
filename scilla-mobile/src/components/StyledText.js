// @flow
import React from 'react';
import { Text, Label } from "native-base"; 
import Fonts from "../constants/Fonts";

export class MonoText extends React.Component<any, any> {
  render() {
    return <Text {...this.props} style={[this.props.style, { fontFamily: Fonts.SpaceMono }]} />;
  }
}

export class AppText extends React.Component<any, any> {
  render() {
    return <Text {...this.props} style={[this.props.style, { fontFamily: Fonts.OpenSans }]} />;
  }
}

export class AppHeaderText extends React.Component<any, any> {
  render() {
    return <Text {...this.props} style={[this.props.style, { fontFamily: Fonts.OpenSans }]} />;
  }
}
