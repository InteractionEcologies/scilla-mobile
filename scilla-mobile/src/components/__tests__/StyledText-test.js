// @flow
import {} from 'react-native';
import React from 'react';
import { MonoText, AppText } from '../StyledText';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<AppText>Snapshot test!</AppText>).toJSON();

  expect(tree).toMatchSnapshot();
});
