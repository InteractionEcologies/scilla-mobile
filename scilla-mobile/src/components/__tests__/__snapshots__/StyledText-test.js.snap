// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "open-sans",
        "fontSize": 15,
      },
    ]
  }
  uppercase={false}
>
  Snapshot test!
</Text>
`;
